db/migrate/20190511150503_decrypt_two_factor_secret.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+class DecryptTwoFactorSecret < ActiveRecord::Migration[5.1]
+  class User < ApplicationRecord
+  end
+
+  def up
+    add_column :users, :plain_otp_secret, :string
+
+    key = twofa_encryption_key
+    decrypt_existing_secrets(key) if key
+
+    change_table :users, bulk: true do |t|
+      t.remove :encrypted_otp_secret
+      t.remove :encrypted_otp_secret_iv
+      t.remove :encrypted_otp_secret_salt
+    end
+  end
+
+  def down
+    raise ActiveRecord::IrreversibleMigration
+  end
+
+  private
+
+  def twofa_encryption_key
+    if AppConfig.heroku?
+      ENV["TWOFA_ENCRYPTION_KEY"]
+    else
+      key_file = File.expand_path("../../config/initializers/twofa_encryption_key.rb", File.dirname(__FILE__))
+
+      if File.exist? key_file
+        require key_file
+        File.delete(key_file)
```

Comment:
Uh, kinda dangerous... Well I'm too late :D

Reply:
Well, I thought if I should make it safer (wait with deletion until after the successful migration, but couldn't find a nice solution without making everything a lot more complex). But since this only affects pods which were on `develop` (since this was never in a release), it shouldn't affect many pods (most pods are probably already upgraded), and if it really fails, their podmins should be able to fix it in the DB and tell the few people who already enabled 2fa to re-enable it (or maybe they even have a backup of the config folder since there are other important files in there too).